### PR TITLE
WIP: Add optional `code` to PUT /resource

### PR DESCRIPTION
**Pull request by Minivera**
Changed the specs for supporting and optional 'import-code' parameter.
Refactored the client to use the model rather than a chain of argument to make future changes easier
Changed tests and acceptance tests to test the import code if any

Work is related to manifoldco/engineering#4580

### CODE REVIEW

File: cmd/test.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ func init() {
 				Usage:  "The label of the region which the resource will be provision in",
 				EnvVar: "REGION",
 			},
+			cli.StringFlag{
+				Name:   "import-code",
+				Usage:  "The import code to import an existing resource for that resource",
+				EnvVar: "REGION",
```
Comment (luizbranco) [inline cmd/test.go]: EnvVar should be `IMPORT_CODE`

File: specs/connector.yaml [hunk 2]
```diff
@@ -470,6 +470,12 @@ definitions:
       Combination of the cloud platform and location to provision this
       resource within.
     pattern: ^([a-z0-9][a-z0-9\-_]{1,63})::([a-z0-9][a-z0-9\-_]{1,63})$
+  ImportCode:
```
Comment (luizbranco) [inline specs/connector.yaml]: @tim-speed Do we need to return this info from the connector? Is there a reason for providers to retrieve the code after importing a resource?

File: cmd/test.go (hunk 1 above)
Comment (Minivera) [reply cmd/test.go]: Good catch, I changed it in the latest commit

File: specs/connector.yaml (hunk 2 above)
Comment (tim-speed) [reply specs/connector.yaml]: Hrmm nope, and we wont be storing it. Should just be in the Provider spec.

Comment (tim-speed) [reply specs/connector.yaml]: We wont be storing it on the resource at least. It will still be in the provision operation

Comment (tim-speed) [reply specs/connector.yaml]: HRMmmm or wait, maybe we should store it on the resource as an indicator that the resource was imported?
What do you think @luizbranco  ? or should we just make it a boolean on the resource

Comment (luizbranco) [reply specs/connector.yaml]: I'm fine saving the code with the resource and for the client we turn into a bool. What do you think? But I don't think we need to change the connector, providers should already have the mapping on their side.

Comment (tim-speed) [reply specs/connector.yaml]: :+1: works for me.

**Pull request merged**